pageserver/src/tenant/timeline.rs

```diff
@@ -776,6 +776,20 @@ pub(crate) enum CompactFlags {
     DryRun,
 }
 
+#[serde_with::serde_as]
+#[derive(Debug, Clone, serde::Deserialize)]
+pub(crate) struct CompactRequest {
+    pub compact_range: Option<CompactRange>,
+    pub compact_below_lsn: Option<LsnDisplay>,
+    /// Whether the compaction job should be scheduled.
+    #[serde(default)]
+    pub scheduled: bool,
+}
+
+#[serde_with::serde_as]
+#[derive(Debug, Clone, serde::Deserialize)]
+pub(crate) struct LsnDisplay(#[serde_as(as = "serde_with::DisplayFromStr")] pub Lsn);
+
```

Comment:
I think this is unnecessary. The `Serialize` impl of `Lsn` already uses the hex repr for human readable formats.